data/org.freedesktop.portal.Notification.xml

```diff
@@ -116,8 +166,211 @@
           * ``target`` (``v``)
 
             Target parameter to send along when activating the action.
+
+          * ``purpose`` (``s``)
+
+            The ``purpose`` of the button. This information may be used by the
+            notification server to style the button specially.
+
+            The standardized hints are defined as part of a ``content-type``.
+            Depending on the ``purpose`` other fields of the button may be ignored.
+            If the server doesn't understand the ``purpose` it will be
+            ignored and the button will be shown as a normal button.
+
+            Additional purposes may be defined by notification servers using
+            ``x-vendor`` prefix e.g. ``x-gnome.class.specific``
+
+        * ``actions`` (``aa{sv}``)
+
+          Array of actions that the notification server will invoke
+          automatically in specific cases. These actions may be called never or even multiple times.
+
+          Invoked actions will contain as parameter a vardict ``a{sv}`` containing the
+          ``target` and the response of the action.
+
+          The format for a serialized action is a vardict with the following supported keys:
+
+          * ``action`` (``s``)
+
+            Name of the action. Mandatory.
+
+            This action needs to be exported by the application.
```

Comment:
This means exported via org.freedesktop.Application? If so, could it be written more explicitly, like `This action needs to be exported via the org.freedesktop.Application interface.`?

Reply:
Hmm, actually, it seems to be more complicated looking at the existing backends. If the action starts with `app.` then it should be exported via org.freedesktop.Application (without the `app.` prefix), otherwise via ActionInvoked.

https://github.com/flatpak/xdg-desktop-portal-gtk/blob/main/src/notification.c#L119-L159
https://invent.kde.org/plasma/xdg-desktop-portal-kde/-/blob/master/src/notification.cpp?ref_type=heads#L102-115

That doesn't seem to be mentioned anywhere and I remember the KDE backend to not work with GTK applications until I reversed what GTK backend does with Bustle and reported to them.

Reply:
It's actually mentioned on top of the notification API. I also added more information about it.